Source/Diffusion.cpp

```diff
@@ -470,9 +470,10 @@ Diffusion::diffuse_scalar (const Vector<MultiFab*>&  S_old,
           auto const& rhs      = Rhs.array(Smfi);
           auto const& solution = Soln.array(Smfi);
           auto const& snew     = S_new[0]->array(Smfi,sigma);
-          auto const& sold     = S_old[0]->array(Smfi,sigma);
-          auto const& rhoHalf  = (rho_flag == 1) ? rho_half.array(Smfi) : Soln.array(Smfi);
-          auto const& rho_old  = Rho_old[0]->array(Smfi,Rho_comp);
+	  Array4<Real> dummy;
+          auto const& sold     = (solve_mode == PREDICTOR) ? S_old[0]->array(Smfi,sigma) : dummy; 
+          auto const& rhoHalf  = rho_half.array(Smfi);
+          auto const& rho_old  = (rho_flag ==3 ) ? Rho_old[0]->array(Smfi,Rho_comp) : dummy;
```

Comment:
@esclapez Is this consistent with your latest rearrangements?

Reply:
That looks good! The dummy Array4 makes you intentions clearer rather than pointing to Soln and rho_half being always available, the test with (rho_flag == 1) was not necessary.